From 85284a44305c89ecbe1dce7aa9d20651db00f066 Mon Sep 17 00:00:00 2001
From: David Mercado <david.mercado@windriver.com>
Date: Tue, 19 Nov 2013 18:38:09 -0500
Subject: [PATCH 138/175] LSI AXM55XX: Enable coherent DMA for USB device

git.yoctoproject.org/git/linux-yocto-3.10
commit 0e7cc67b4c0d79769fe70ffc9aa94b757887594f standard/axxia/base.

Change dma ops to use the coherent versions on devices with the 'dma-coherent'
attribute in the device tree.

Signed-off-by: Anders Berg <anders.berg@lsi.com>
---
 arch/arm/boot/dts/axm55xx.dts   |    1 +
 drivers/usb/host/ehci-ci13612.c |    4 ----
 2 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/axm55xx.dts b/arch/arm/boot/dts/axm55xx.dts
index 896cb7f..f84d909 100644
--- a/arch/arm/boot/dts/axm55xx.dts
+++ b/arch/arm/boot/dts/axm55xx.dts
@@ -337,6 +337,7 @@
 		reg = <0x20 0x10140000 0x0 0020000>,
 		      <0x20 0x10094000 0x0 0002000>;
 		interrupts = <0 55 4>;
+		dma-coherent = <1>;
 	};
 
 	amba {
diff --git a/drivers/usb/host/ehci-ci13612.c b/drivers/usb/host/ehci-ci13612.c
index 2a30dab..4c86e9b 100644
--- a/drivers/usb/host/ehci-ci13612.c
+++ b/drivers/usb/host/ehci-ci13612.c
@@ -284,10 +284,6 @@ static int ci13612_ehci_probe(struct platform_device *pdev)
 	} else {
 		/* Set address bits [39:32] to zero */
 		writel(0x0, gpreg_base + 0x8);
-#ifndef CONFIG_LSI_USB_SW_WORKAROUND
-		/* hprot pass-through (let the controller drive hprot[0:3] */
-		writel(0x100, gpreg_base + 0x74);
-#endif
 		iounmap(gpreg_base);
 	}
 
-- 
1.7.1

